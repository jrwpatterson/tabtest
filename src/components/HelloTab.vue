<template>
  <div class="container">
    <div>Results</div>
    <div class="ball-container" v-if="powerData.PrimaryNumbers.length">
      <Balls
        v-bind:key="item"
        v-for="item in powerData.PrimaryNumbers"
        v-bind:value="item"
      />
      <Balls
        v-bind:value="powerData.SecondaryNumbers[0]"
        v-bind:isPowerBall="true"
      />
    </div>
    <div v-else>
      <button data-testid="get.results" @click="dataFetch">
        Auto Fill Results
      </button>
      no data
    </div>

    <TicketGrid
      v-bind:powerBalls="powerData.SecondaryNumbers"
      v-bind:ticketNumbers="powerData.PrimaryNumbers"
    />
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import axios from "axios";
import _ from "lodash";
import Balls from "./Balls.vue";
import TicketGrid from "./TicketGrid.vue";
import { DataProps, LotteryData, DrawResult } from "./interfaces";

export default Vue.extend({
  name: "HelloWorld",
  props: {
    msg: String
  },
  data: function(): DataProps {
    return {
      powerData: {
        PrimaryNumbers: [],
        SecondaryNumbers: []
      }
    };
  },
  components: {
    Balls,
    TicketGrid
  },
  methods: {
    async dataFetch() {
      const lotteryData = await axios.post<LotteryData>(
        "https://data.api.thelott.com/sales/vmax/web/data/lotto/latestresults",
        {
          CompanyId: "GoldenCasket",
          MaxDrawCountPerProduct: 1,
          OptionalProductFiler: ["Powerball"]
        }
      );
      this.powerData = _.find(
        lotteryData.data.DrawResults,
        (result: DrawResult) => result.ProductId === "Powerball"
      );
    }
  }
});
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
.ball-container {
  display: flex;
  flex-direction: row;
  justify-content: center;
}
.container {
  display: flex;
  flex-direction: column;
  align-content: center;
}
.aligner {
  display: flex;
  flex-direction: column;
  align-content: center;
  width: 320px;
  align-self: center;
}
</style>
