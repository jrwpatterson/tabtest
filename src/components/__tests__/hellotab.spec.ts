import { render, fireEvent } from "@testing-library/vue";
import HelloTab from "../HelloTab.vue";
import axiosMock from "axios";
import { LotteryData } from "../interfaces";
import "@testing-library/jest-dom";

const mockResponst: LotteryData = {
  ErrorInfo: null,
  Success: true,
  DrawResults: [
    {
      ProductId: "Powerball",
      PrimaryNumbers: [1, 15, 25, 26, 27, 32, 33],
      SecondaryNumbers: [13]
    }
  ]
};

describe("hello tab home page", () => {
  it("should match snapshot", () => {
    const { container } = render(HelloTab);
    expect(container).toMatchSnapshot();
  });

  it("should match snapshot when clicked", async () => {
    ((axiosMock.post as unknown) as jest.Mock).mockImplementationOnce(() =>
      Promise.resolve({
        data: mockResponst
      })
    );

    const { getByTestId, container } = render(HelloTab);

    fireEvent.click(getByTestId("get.results"));
    expect(container).toMatchSnapshot();
  });

  it("button should call the server", async () => {
    ((axiosMock.post as unknown) as jest.Mock).mockImplementationOnce(() =>
      Promise.resolve({
        data: mockResponst
      })
    );

    const { getByTestId } = render(HelloTab);

    expect(getByTestId("get.results")).toBeVisible();
    fireEvent.click(getByTestId("get.results"));
    await Promise.resolve();
    expect(axiosMock.post).toHaveBeenCalled();
  });
});
