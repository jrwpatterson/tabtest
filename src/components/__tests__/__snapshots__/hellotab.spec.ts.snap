// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello tab home page should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Results
    </div>
     
    <div>
      <button
        data-testid="get.results"
      >
        
      Auto Fill Results
    
      </button>
      
    no data
  
    </div>
     
    <div
      class="ticket-grid"
    >
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            1
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            2
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            3
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            4
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            5
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            6
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            7
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            8
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            9
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            10
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            11
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            12
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            13
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            14
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            15
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            16
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            17
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            18
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            19
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            20
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            21
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            22
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            23
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            24
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            25
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            26
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            27
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            28
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            29
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            30
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            31
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            32
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            33
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            34
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            35
          </div>
        </div>
      </div>
       
      <div
        class="select-powerball"
      >
        SELECT YOUR POWERBALL
      </div>
       
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            1
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            2
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            3
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            4
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            5
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            6
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            7
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            8
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            9
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            10
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            11
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            12
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            13
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            14
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            15
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            16
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            17
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            18
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            19
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            20
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`hello tab home page should match snapshot when clicked 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      Results
    </div>
     
    <div>
      <button
        data-testid="get.results"
      >
        
      Auto Fill Results
    
      </button>
      
    no data
  
    </div>
     
    <div
      class="ticket-grid"
    >
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            1
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            2
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            3
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            4
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            5
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            6
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            7
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            8
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            9
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            10
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            11
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            12
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            13
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            14
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            15
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            16
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            17
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            18
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            19
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            20
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            21
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            22
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            23
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            24
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            25
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            26
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            27
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            28
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            29
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            30
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            31
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            32
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            33
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            34
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            35
          </div>
        </div>
      </div>
       
      <div
        class="select-powerball"
      >
        SELECT YOUR POWERBALL
      </div>
       
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            1
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            2
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            3
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            4
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            5
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            6
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            7
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            8
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            9
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            10
          </div>
        </div>
      </div>
      <div
        class="row-outer"
      >
        <div>
          <div
            class="number-container"
          >
            11
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            12
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            13
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            14
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            15
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            16
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            17
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            18
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            19
          </div>
        </div>
        <div>
          <div
            class="number-container"
          >
            20
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
